src/main/java/uk/co/real_logic/benchmarks/rtt/SampleMessageProvider.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2015-2020 Real Logic Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.co.real_logic.benchmarks.rtt;
+
+import sun.misc.Unsafe;
+
+import java.util.Arrays;
+
+import static org.agrona.BitUtil.CACHE_LINE_LENGTH;
+import static org.agrona.BitUtil.SIZE_OF_LONG;
+import static org.agrona.UnsafeAccess.UNSAFE;
+
+public final class SampleMessageProvider implements MessageProvider
```

Comment:
Reflecting on this a possible name is "MessagePump".

Reply:
Done